### PR TITLE
feat: Added max_instance_lifetime option for Auto Scaling Group

### PR DESCRIPTION
**Pull request by zahorniak**
## Description
I've added missed [max_instance_lifetime](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#max_instance_lifetime) option for Auto Scaling Group.

### CODE REVIEW

**Comment by zahorniak** [comment]
@antonbabenko `versions.tf` added. The minimal allowed version of Terraform AWS provider is 2.41.0.

Tools version:
```
$ terraform -v
Terraform v0.12.24
$ terraform-docs -v
terraform-docs version v0.0.0- v0.9.1 darwin/amd64 BuildDate: 2020-04-03T02:30:49+0100
```

File: versions.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = "~> 0.12.6"
+
+  required_providers {
+    aws    = "~> 2.41.0"
+    random = "~> 2.2.1"
```
Comment (antonbabenko) [inline versions.tf:6]: 1. Remove `random` and `null` from this list.
2. Replace `aws    = "~> 2.41.0"` with `aws    = "~> 2.41"`

PS: Make it exactly as in VPC module.

**Comment by zahorniak** [comment]
@antonbabenko fixed

**Comment by antonbabenko** [comment]
Thanks @zahorniak !

[v3.5.0](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/releases/v3.5.0) has been just released.

**Pull request merged**